hvantk/annotation/annotation_pipeline.py

```
@@ -0,0 +1,435 @@
+            # Auto-detect format and load
+            data = self._auto_load_data(self.config.source_path)
+
+        # Apply preprocessing if specified
+        if self.config.preprocessing_func:
+            data = self.config.preprocessing_func(data)
+
+        return data
+
+    def _auto_load_data(self, path: str) -> hl.Table:
+        """Automatically detect file format and load appropriately"""
+        path_obj = Path(path)
```

Ruff (0.12.2)

Undefined name `Path`

(F821)